WEB/Src/WindowsServer/WindowsServer/Implementation/AzureMetadataRequestor.cs

```diff
@@ -58,52 +52,40 @@ public Task<AzureInstanceComputeMetadata> GetAzureComputeMetadataAsync()
 
         private async Task<AzureInstanceComputeMetadata> MakeAzureMetadataRequestAsync(string metadataRequestUrl)
         {
-            AzureInstanceComputeMetadata requestResult = null;
-
             SdkInternalOperationsMonitor.Enter();
             try
             {
-                if (this.azureIMSRequestor != null)
-                {
-                    requestResult = await this.azureIMSRequestor(metadataRequestUrl).ConfigureAwait(false);
-                }
-                else
-                {
-                    requestResult = await this.MakeWebRequestAsync(metadataRequestUrl).ConfigureAwait(false);
-                }
+                return await this.MakeWebRequestAsync(metadataRequestUrl).ConfigureAwait(false);
             }
             catch (Exception ex)
             {
                 WindowsServerEventSource.Log.AzureInstanceMetadataRequestFailure(
                     metadataRequestUrl, ex.Message, ex.InnerException != null ? ex.InnerException.Message : string.Empty);
+
+                return null;
             }
             finally
             {
                 SdkInternalOperationsMonitor.Exit();
             }
-
-            return requestResult;
         }
         
         private async Task<AzureInstanceComputeMetadata> MakeWebRequestAsync(string requestUrl)
         {
             AzureInstanceComputeMetadata azureIms = null;
             DataContractJsonSerializer deserializer = new DataContractJsonSerializer(typeof(AzureInstanceComputeMetadata));
 
-            using (var azureImsClient = new HttpClient())
-            {
-                azureImsClient.MaxResponseContentBufferSize = AzureMetadataRequestor.AzureImsMaxResponseBufferSize;
-                azureImsClient.DefaultRequestHeaders.Add("Metadata", "True");
-                azureImsClient.Timeout = this.AzureImsRequestTimeout;
+            this.httpClient.MaxResponseContentBufferSize = AzureMetadataRequestor.AzureImsMaxResponseBufferSize;
+            this.httpClient.DefaultRequestHeaders.Add("Metadata", "True");
```

Comment:
Would this end up adding multiple headers to the `httpClient`?

Reply:
moved to constructor